### PR TITLE
Split nordic sdk into its own module

### PR DESCRIPTION
**Pull request by LiyouZhou**
Remove nordic files from this module and make ble-nrf51822 depend on nrf51-sdk. @rgrover

### CODE REVIEW

**Comment by andresag01** [comment]
I wonder whether this is a good opportunity to remove the `bootloder/` directory since it should  actually be in the nrf51 "family" targets ([see this pull request](https://github.com/ARMmbed/target-nrf51822/pull/32)). However, if we are going to leave it here, I am unsure whether we need some sort of license file for it.

**Comment by rgrover** [comment]
ble-nrf51822 is also shared with mbed-classic. The mbed-classic toolchain expects to find the bootloader here.

**Comment by andresag01** [comment]
Ah ok, forgot about that

File: LICENSE [hunk 1]
```diff
@@ -1,9 +1,7 @@
-Many of the files in this module have been inherited from the Nordic SDK for
-nRF51822. With the exception of the softdevice, they come with a BSD license
+This module contains softdevice which comes with a BSD license
```
Comment (rgrover) [inline LICENSE]: you've got two sentences here which are both talking about the softdevice license. The softdevice is covered by the Softdevice License Agreement.

File: module.json [hunk 2]
```diff
@@ -24,29 +24,13 @@
     }
   ],
   "dependencies": {
-    "ble": "^2.0.0"
+    "ble": "^2.0.0",
+    "nrf51-sdk": "*"
```
Comment (rgrover) [inline module.json]: would be nice to say which version of the nrf51-sdk we depended on. ^8.0.0 perhaps.

**Comment by rgrover** [comment]
happy with it in general.

**Comment by LiyouZhou** [comment]
@rgrover please review.

**Comment by rgrover** [comment]
ready for merge. I'll now check if the sources in nrf51-sdk match the files we're losing on ble-nrf51dk.
What testing have you performed?

**Comment by LiyouZhou** [comment]
@rgrover broke ble-examples. hold off merging pending further investigation.

**Comment by metc** [comment]
@LiyouZhou IMO, splitting the Nordik SDK to its own module is great !

Should the `nRF51` SDK be modified/patched from the original sources available here ? http://developer.nordicsemi.com/nRF51_SDK/nRF51_SDK_v8.x.x/

**Comment by metc** [comment]
Answer here, sorry... https://github.com/ARMmbed/nrf51-sdk/blob/master/README.md#changes-made-to-nordic-files

**Comment by LiyouZhou** [comment]
@rgrover I have made the fixes required. Please continue reviewing this merge.

**Comment by rgrover** [comment]
this a large collection of deletions, which is what I had expected it to be. is the nrf51-sdk module populated with exactly the same files which have been deleted from here?

**Comment by LiyouZhou** [comment]
@rgrover yes and more files used in other modules.

**Comment by rgrover** [comment]
published ble-nrf51822-2.2.0

**Pull request merged**